src/main/java/net/minecraftforge/common/PlantType.java

```diff
@@ -19,27 +19,71 @@
 
 package net.minecraftforge.common;
 
+import net.minecraft.block.Block;
 import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.GlazedTerracottaBlock;
+import net.minecraft.block.material.Material;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.tags.BlockTags;
 import net.minecraft.util.Direction;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.IBlockReader;
 
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
 import java.util.regex.Pattern;
 
 public final class PlantType
 {
     private static final Pattern INVALID_CHARACTERS = Pattern.compile("[^a-z_]"); //Only a-z and _ are allowed, meaning names must be lower case. And use _ to separate words.
     private static final Map<String, PlantType> VALUES = new ConcurrentHashMap<>();
 
-    public static final PlantType PLAINS = get("plains");
-    public static final PlantType DESERT = get("desert");
-    public static final PlantType BEACH = get("beach");
-    public static final PlantType CAVE = get("cave");
-    public static final PlantType WATER = get("water");
-    public static final PlantType NETHER = get("nether");
-    public static final PlantType CROP = get("crop");
+    public static final PlantType PLAINS = get("plains", x -> x.getState().is(Blocks.GRASS_BLOCK) || Tags.Blocks.DIRT.contains(x.getState().getBlock()) || x.getState().is(Blocks.FARMLAND));
```

Comment:
`x` is confusing here. Generally, lambda parameters are named as abbreviated versions of their otherwise normal parameters. I would prefer something like `pi` for `plantInfo`.